drivers/SmartThings/matter-appliance/src/matter-refrigerator/init.lua

```diff
@@ -96,15 +96,43 @@ local function is_matter_refrigerator(opts, driver, device)
   return false
 end
 
-local function temperature_setpoint_attr_handler(driver, device, ib, response)
-  local tn_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID, {feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER})
+local function supports_temperature_level_endpoint(device, endpoint)
```

Comment:
I don't think the `supports_temperature_level_endpoint` and `supports_temperature_number_endpoint` functions are needed, see my explanation in the comments below

Reply:
I see that these checks are used throughout the file, but I don't think they are needed. Have you seen many instances where an attribute handler has been called for an endpoint that doesn't support the correct feature (temperatureNumber or temperatureLevel). That seems like more of an `endpoint_to_component` map issue, but I think that should be addressed with the `COMPONENT_TO_ENDPOINT` field that is used in this driver to map the components to the correct endpoints.

Reply:
@ctowns 
I checked that handler for unsupported feature(TN or TL) is called in Virtual Matter App.  I think the server can call the handler for features that are not supported. so we need to prevent this case.